tests/basic/pool_function_test.go

```diff
@@ -388,3 +389,103 @@ var _ = Describe("{PoolExpansionDiskResizeInvalidSize}", func() {
 	})
 
 })
+
+var _ = Describe("{PoolExpandResizeWithSameSize}", func() {
+
+	BeforeEach(func() {
+		StartTorpedoTest("PoolExpandResizeWithSameSize",
+			"Initiate pool expansion using same size", nil, testrailID)
+	})
+
+	AfterEach(func() {
+		EndTorpedoTest()
+	})
+
+	stepLog := "select a pool and expand it by same pool size with resize-disk type"
+	It(stepLog, func() {
+		log.InfoD(stepLog)
+		// pick pool to resize
+		pools, err := GetAllPoolsPresent()
+		log.FailOnError(err, "Unable to get the storage Pools")
+		pooltoPick := pools[0]
+		poolToBeResized = getStoragePool(pooltoPick)
+
+		originalSizeGiB := poolToBeResized.TotalSize / units.GiB
+		targetSizeGiB = originalSizeGiB
+		resizeErr := Inst().V.ExpandPool(pooltoPick, api.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK, targetSizeGiB, true)
+		dash.VerifyFatal(resizeErr != nil, true, "Verify error occurs with same pool size")
+
+		// Verify error on pool expansion failure
+		var errMatch error
+		re := regexp.MustCompile(`.*already at a size.*`)
+		if !re.MatchString(fmt.Sprintf("%v", resizeErr)) {
+			errMatch = fmt.Errorf("failed to verify failure using same Pool size")
+		}
+		dash.VerifyFatal(errMatch, nil, "Pool expand with Same Pool Size completed?")
+	})
+})
+
+var _ = Describe("{PoolExpandWhileResizeDiskInProgress}", func() {
+
+	BeforeEach(func() {
+		StartTorpedoTest("PoolExpandWhileResizeDiskInProgress",
+			"Initiate pool expansion on a pool where one pool expansion is already in progress", nil, testrailID)
+		contexts = scheduleApps()
+	})
+
+	JustBeforeEach(func() {
+		poolIDToResize = pickPoolToResize()
+		log.Infof("Picked pool %s to resize", poolIDToResize)
+		poolToBeResized = getStoragePool(poolIDToResize)
+		storageNode, err = GetNodeWithGivenPoolID(poolIDToResize)
+		log.FailOnError(err, "Failed to get node with given pool ID")
+	})
+
+	JustAfterEach(func() {
+		AfterEachTest(contexts)
+	})
+
+	AfterEach(func() {
+		appsValidateAndDestroy(contexts)
+		EndTorpedoTest()
+	})
+
+	stepLog := "should get the existing pool and expand it by adding a resize-disk and again trigger pool expand on same pool"
+	It(stepLog, func() {
+		log.InfoD(stepLog)
+
+		originalSizeInBytes = poolToBeResized.TotalSize
+		targetSizeInBytes = originalSizeInBytes + 100*units.GiB
+		targetSizeGiB = targetSizeInBytes / units.GiB
+
+		log.InfoD("Current Size of the pool %s is %d GiB. Trying to expand to %v GiB with type resize-disk",
+			poolIDToResize, poolToBeResized.TotalSize/units.GiB, targetSizeGiB)
+		triggerPoolExpansion(poolIDToResize, targetSizeGiB, api.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK)
+
+		pxctlCmdFull := fmt.Sprintf("pxctl sv pool expand -u %s -s %d -o resize-disk ", poolIDToResize, targetSizeGiB)
```

Comment:
Why dont we use the same `triggerPoolExpansion` or at least ` Inst().V.ExpandPool()` API call instead of using CLI command here?

I see, in attached Jenkins job run link, job was not able to SSH to the node, to perform this CLI command.
```
12:16:41 2023-10-24 06:46:41 +0000:[INFO] [{PoolExpandWhileResizeDiskInProgress}] [ssh.(*SSH).doCmdSSH:#696] - SSH ERR: Process exited with status 1
```

Reply:
we need to use cuz we don't want retries here

Reply:
And the pxctl command fails with the error